sympy/geometry/ellipse.py

```diff
@@ -1362,6 +1357,46 @@ def __new__(cls, *args, **kwargs):
 
         raise GeometryError("Circle.__new__ received unknown arguments")
 
+    def object_from_equation(self):
+        """Returns a Circle object from an equation of circle given as input.
+        If the equation of the circle is ax**2 + by**2 + gx + hy + c = 0, then
+        the input has to be of the following format: ax**2 + by**2 + gx + hy + c
+
+        Examples
+        ========
+
+        >>> from sympy import Circle
+        >>> from sympy.abc import x, y
+        >>> Circle.object_from_equation(x**2 + y**2 + 3*x + 4*y - 8)
+        Circle(Point2D(-3/2, -2), 3.77491721763537)
+
+        """
+
+        from sympy.abc import x, y
+        poly_x = poly(self, x)
+        poly_y = poly(self, y)
+        deg_x = degree(poly_x)
+        deg_y = degree(poly_y)
+        lead_x_coeff = LC(poly_x)
+        lead_y_coeff = LC(poly_y)
+
+        if deg_x == 2 and deg_y == 2:
+            if lead_x_coeff == lead_y_coeff:
```

Comment:
These conditions could be clubbed `deg_x == deg_y == 2 and lead_x_coeff == lead_y_coeff` and correspondingly have a single statement for raising exception (if it is not the case). Also, did you consider degenerate cases?

Reply:
@sidhantnagpal  
>These conditions could be clubbed deg_x == deg_y == 2 and lead_x_coeff == lead_y_coeff and correspondingly have a single statement for raising exception (if it is not the case).

I will do that
>Also, did you consider degenerate cases?

Could you elaborate what you mean by `degenerate` cases?

Reply:
Point circle case. (0 radius)

Reply:
@sidhantnagpal Yes, it works for point circle cases.
```
avi@avi-Aspire-A515-51G:~/sympy$ python3
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import Circle
>>> from sympy.abc import *
>>> Circle.object_from_equation(x**2 + y**2 + 6*x + 8*y + 25)
Circle(Point2D(-3, -4), 0)
```
I could make it return a `GeometryError` for point circle cases too, if you recommend. What do you think? Should that be done or is it fine like the way it is now?